### PR TITLE
5.x Fix secure postLink() data post

### LINKED ISSUE #18223: PostLink Form Protection false positive
### Description

> Tampered field `enable` in POST data (expected value `1` but found `1`)

```
<p><?php echo $this->Form->postLink('Create token', ['action' => 'index'], ['data' => ['enable' => true], 'class' => 'btn btn-primary'])?></p>
```
and CORE/src/Controller/Component/FormProtectionComponent.php at line 156 fires

It seems to work with integer one, but booleans IMO should also work.

### CakePHP Version

5.1

### PHP Version

8.3

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/cakephp/cakephp/issues/18223

Not sure how a test case would look like

I tried

```php
    public function testFormSecuredControlPostLink(): void
    {
        $this->View->setRequest($this->View->getRequest()
            ->withAttribute('formTokenData', [])
            ->withAttribute('csrfToken', 'testKey'));

        $options = [
            'data' => ['string' => 'value', 'boolean' => true],
        ];
        $this->Form->postLink('title', '/articles/add', $options);

        //$this->Form->create();
        $data = $options['data'] + [
            '_Token' => $this->Form->getFormProtector()->buildTokenData(),
        ];

        $this->assertTrue($this->Form->getFormProtector()->validate($data, '', ''));
    }
```
but that doesnt quite work out.

### CODE REVIEW

**Comment by markstory** [comment]
Added a test based on what you had proposed. I had to use some HTML/string munging but it wasn't too messy.

**Comment by ADmad** [comment]
Should add a test for `false` also to show that you will get an empty string for it.

File: src/Form/FormProtector.php [hunk 1]
```diff
@@ -390,9 +390,12 @@ public function buildTokenData(string $url = '', string $sessionId = ''): array
 
         $locked = [];
         foreach ($fields as $key => $value) {
-            if (is_numeric($value)) {
+            if (is_bool($value)) {
+                $value = (string)$value;
+            } elseif (is_numeric($value)) {
```
Comment (ADmad) [inline src/Form/FormProtector.php]:
```suggestion
            if (is_numeric($value) || is_bool($value)) {
```

**Comment by markstory** [comment]
> Should add a test for false also to show that you will get an empty string for it.

Good thinking, there was a bug where the data generated by the HTML array would not match the data used in the signature.

**Review by ADmad — APPROVED** [review]

**Comment by ADmad** [comment]
> Good thinking, there was a bug where the data generated by the HTML array would not match the data used in the signature.

I had a feeling that coercing `false` to an empty string could be an issue :)
